src/include/alloc.h

```diff
@@ -219,6 +219,11 @@ static inline ncclResult_t ncclCuMemAllocAddr(void **ptr, CUmemGenericAllocation
   accessDesc.flags = CU_MEM_ACCESS_FLAGS_PROT_READWRITE;
   CUCHECK(cuMemSetAccess((CUdeviceptr)*ptr, size, &accessDesc, 1));
   TRACE(NCCL_ALLOC, "CuMem Map Size %zu pointer %p handle %llx", size, *ptr, *handleIn);
+  if (cudaDev < MAX_ALLOC_TRACK_NGPU) {
```

Comment:
Since logging is tied up to `NCCL_ALLOC` level, there's no need to track if ncclDebugLevel < NCCL_ALLOC. Can you guard the alloc tracker by debugLevel cond everywhere?